### PR TITLE
Profile default avatars

### PR DESCRIPTION
**Pull request by eoger**
Part of #461

### CODE REVIEW

File: components/fxa-client/src/ffi.rs [hunk 1]
```diff
@@ -125,6 +125,7 @@ pub struct ProfileC {
     uid: *mut c_char,
     email: *mut c_char,
     avatar: *mut c_char,
+    avatar_default: bool,
```
Comment (vladikoff) [inline components/fxa-client/src/ffi.rs]: 👍

File: components/fxa-client/src/ffi.rs [hunk 2]
```diff
@@ -145,6 +146,7 @@ impl From<Profile> for ProfileC {
             uid: rust_string_to_c(profile.uid),
             email: rust_string_to_c(profile.email),
             avatar: rust_string_to_c(profile.avatar),
+            avatar_default: profile.avatar_default,
```
Comment (vladikoff) [inline components/fxa-client/src/ffi.rs]: this is nice, no need to drop or anything for the boolean in the ffi 👍

File: components/fxa-client/src/ffi.rs (hunk 1 above)
Comment (thomcc) [reply components/fxa-client/src/ffi.rs]: Don't use bool for things exposed to JNA, it has a bug where it thinks bools are 32 bits.

Comment (vladikoff) [reply components/fxa-client/src/ffi.rs]: gosh, not even booleans are safe :((((

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,16 @@
...
 
+## FxA
+
+### What's New
+
+- It is now possible to know whether a profile avatar has been set by the user. ([#461](https://github.com/mozilla/application-services/issues/461))
+
+### Breaking Changes
+
+- The `avatar` accessor from the `Profile` class in the Swift framework now returns an optional `Avatar` struct instead of a `String`.
```
Comment (thomcc) [inline CHANGELOG.md]: nit: bug number/link

**Comment by rfk** [comment]
How will we track/do the upstream a-c API change, and do we need to resolve any questions around it before merging a breaking API change to master here?

(I'm specifically thinking of the period where we couldn't make a new a-s release for a while because we'd landed some breaking FxA API changes but not yet reflected them into a-c, and what we can do to avoid that in future)

**Comment by thomcc** [comment]
Had I known this was incoming I would have waited to include it in 0.14.0 (updating in https://github.com/mozilla-mobile/android-components/pull/1765). But since that should land soon, there's no real problem since nothing else should land thats a breaking change really.

**Comment by eoger** [comment]
@thomcc might reviewing again?
Ran the u8 changes against the iOS and Android sample apps and the boolean was set correctly.

**Pull request merged**